dot/network/host.go

```diff
@@ -264,32 +264,28 @@ func (h *host) bootstrap() {
 
 // send writes the given message to the outbound message stream for the given
 // peer (gets the already opened outbound message stream or opens a new one).
-func (h *host) send(p peer.ID, pid protocol.ID, msg Message) (err error) {
-	// get outbound stream for given peer
-	s := h.getOutboundStream(p, pid)
+func (h *host) send(p peer.ID, pid protocol.ID, msg Message, stream libp2pnetwork.Stream) error {
```

Comment:
why does this function need to be changed? generally if we know about the stream we can use `writeToStream`, if we're opening it for the first time we can use `send`

Reply:
also this won't this panic if `stream` is nil which it seems to be in the tests? I would recommend changing it back